File: angular/src/app/game/player.ts
import { Signal, computed, signal } from "@angular/core"
import type { Hand } from "./hand"

export class Player {
    public readonly hand = signal(undefined as Hand | undefined)

    public readonly score = signal(0)
    public readonly wonLastGame = signal(false)
    public readonly lostLastGame = signal(false)

    public readonly percentage: Signal<string>

    constructor(
        gamesPlayed: Signal<number>
    ) {
        this.percentage = computed(() =>
            gamesPlayed() ? Math.round(this.score() * 100 / gamesPlayed()) + '%' : '0%')
    }

    public win() {
        this.score.set(this.score() + 1)
        this.wonLastGame.set(true)
        this.lostLastGame.set(false)
    }

    public lose() {
        this.wonLastGame.set(false)
        this.lostLastGame.set(true)
    }

    public tie() {
        this.wonLastGame.set(false)
        this.lostLastGame.set(false)
    }
}

File: blazor/Player.cs
namespace Demo
{
    public class Player
    {
        public Hand? Hand = null;

        public int score = 0;
        public bool wonLastGame = false;
        public bool lostLastGame = false;

        private readonly Score gamesPlayed;

        public string Percentage
        {
            get
            {
                return 0 < this.gamesPlayed.Count
                 ? Math.Round(this.score * 100.0 / gamesPlayed.Count) + "%"
                 : "0%";
            }
        }

        public Player(Score gamesPlayed)
        {
            this.gamesPlayed = gamesPlayed;
        }

        public void win()
        {
            ++this.score;
            this.wonLastGame = true;
            this.lostLastGame = false;
        }

        public void lose()
        {
            this.wonLastGame = false;
            this.lostLastGame = true;
        }

        public void tie()
        {
            this.wonLastGame = false;
            this.lostLastGame = false;
        }
    }
}